src/backgroundfilter/BackgroundFilterVideoFrameProcessor.ts

```diff
@@ -0,0 +1,176 @@
+    const {
+      paths = DEFAULT_PATHS,
+      model = CREATE_DEFAULT_MODEL_SPEC(),
+      assetGroup = this.defaultAssetSpec().assetGroup,
+      revisionID = this.defaultAssetSpec().revisionID,
+    } = spec || {};
+
+    const params = {
+      assetGroup,
+      revisionID,
+      sdk: encodeURIComponent(Versioning.sdkVersion),
+      ua: encodeURIComponent(Versioning.sdkUserAgentLowResolution),
+    };
+
+    paths.worker = this.createUrlWithParams(paths.worker, params);
+    paths.wasm = this.createUrlWithParams(paths.wasm, params);
+    paths.simd = this.createUrlWithParams(paths.simd, params);
+    model.path = this.createUrlWithParams(model.path, params);
+
+    return {
+      paths,
+      model,
+      assetGroup,
+      revisionID,
+    };
+  }
+
+  /**
+   * Based on the options that are passed in set defaults for options
+   * @param options  the options that are passed in
+   * @returns An updated set of options with defaults set
+   */
+  protected static resolveOptions(options?: BackgroundFilterOptions): BackgroundFilterOptions {
+    if (!options.reportingPeriodMillis) {
+      options.reportingPeriodMillis = 1000;
+    }
+
+    const DEFAULT_FILTER_CPU_UTILIZATION = 30;
+    if (!options.filterCPUUtilization) {
+      options.filterCPUUtilization = DEFAULT_FILTER_CPU_UTILIZATION;
+    } else if (options.filterCPUUtilization < 0 || options.filterCPUUtilization > 100) {
+      options.logger.warn(
+        `filterCPUUtilization must be set to a range between 0 and 100 percent. Falling back to default of ${DEFAULT_FILTER_CPU_UTILIZATION} percent`
+      );
+      options.filterCPUUtilization = DEFAULT_FILTER_CPU_UTILIZATION;
+    }
+    return options;
+  }
+
+  /**
+   * This method will detect the environment in which it is being used and determine if background
+   * blur/replacement can be used.
+   * @param spec The {@link BackgroundBlurSpec} spec that will be used to initialize assets
+   * @param options options such as logger
+   * @returns a boolean promise that will resolve to true if supported and false if not
+   */
+  static isSupported(spec?: BackgroundFilterSpec, options?: { logger?: Logger }): Promise<boolean> {
+    const { logger } = options;
+
+    // could not figure out how to remove globalThis to test failure case
+    /* istanbul ignore next */
+    if (typeof globalThis === 'undefined') {
+      logger.info('Browser does not have globalThis.');
+      return Promise.resolve(false);
+    }
+
+    const browser = new DefaultBrowserBehavior();
+    if (!browser.supportsBackgroundFilter()) {
+      logger.info('Browser is not supported.');
+      return Promise.resolve(false);
+    }
+
+    if (!supportsWASM(globalThis, logger)) {
+      logger.info('Browser does not support WASM.');
+      return Promise.resolve(false);
+    }
+    return this.supportsBackgroundBlur(globalThis, spec, logger);
+  }
+
+  private static async supportsBackgroundBlur(
```

Comment:
Isn't background replacement also check this? If so, should we just move it to supported?

Reply:
good catch. This was just a miss on refactoring. I will rename it to supportsBackgroundFilter.